### PR TITLE
Fix interval first metrics collection after starting the server

### PR DESCRIPTION
**Pull request by arnaudsjs**
# Description

Ensure that we only write metrics to the database after the server has been up for `COLLECTION_INTERVAL_IN_SEC` seconds.

# Self Check:

- [ ] ~~Attached issue to pull request~~
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] ~~End user documentation is included or an issue is created for end-user documentation~~

### CODE REVIEW

**Review by sanderr — APPROVED** [review]

File: src/inmanta/server/services/environment_metrics_service.py [hunk 1]
```diff
@@ -139,7 +139,10 @@ async def start(self) -> None:
         self.register_metric_collector(ResourceCountMetricsCollector())
         self.register_metric_collector(AgentCountMetricsCollector())
         self.register_metric_collector(CompileTimeMetricsCollector())
-        self.schedule(self.flush_metrics, COLLECTION_INTERVAL_IN_SEC, initial_delay=0, cancel_on_stop=True)
+        self.previous_timestamp = datetime.now()
```
Comment (Hugo-Inmanta) [inline src/inmanta/server/services/environment_metrics_service.py:142]: What's the rationale behind setting this here as well as in the `__init__` ?

**Review by Hugo-Inmanta — APPROVED** [review]

File: src/inmanta/server/services/environment_metrics_service.py (hunk 1 above)
Comment (arnaudsjs) [reply src/inmanta/server/services/environment_metrics_service.py:142]: The one in `__init__` is actually not required. Only the one here is required. But I didn't want to make `self.previous_timestamp` optional, so I set it in both locations.

Comment (arnaudsjs) [reply src/inmanta/server/services/environment_metrics_service.py:142]: Giving it a second though, this line is not really required here. The one in `__init__.py` should be sufficient. I will remove this one.

[the remark this answers is not in the document]
Comment (arnaudsjs) [reply src/inmanta/server/services/environment_metrics_service.py]: You can create a ticket for this.

[the remark this answers is not in the document]
Comment (sanderr) [reply src/inmanta/server/services/environment_metrics_service.py]: #5356

**Pull request closed without merging**